torchtext/experimental/vocab.py

```diff
@@ -1,12 +1,58 @@
+from collections import OrderedDict
 import logging
 from typing import Dict, List
 
 import torch
 import torch.nn as nn
+from tqdm import tqdm
 
 logger = logging.getLogger(__name__)
 
 
+def _infer_shape(f):
+    num_lines = 0
+    for line in f:
+        num_lines += 1
+    f.seek(0)
+    return num_lines
+
+
+def vocab_from_file_object(file_like_object, **kwargs):
+    r"""Create a `Vocab` object from a file like object.
+
+    Note that the tensor corresponding to each vector is of type `torch.float`.
```

Comment:
Can this be a txt file, which is a column of tokens? Or it must be a csv file?

Reply:
This could be a `txt` file. Let me update the docs to reflect that!